### PR TITLE
Add @ember/jquery and @ember/optional-features by default.

### PR DESCRIPTION
**Pull request by rwjblue**
* Apps get `@ember/jquery` for backwards compatibility
* Addons do not include `@ember/jquery`
* All projects include `@ember/optional-features` and either opt-in or
  out as appropriate.
* Update to ember-try 1.0.0-beta.3
* Make default scenario run without jQuery
* Add `default-with-jquery` ember-try scenario

### CODE REVIEW

**Comment by Turbo87** [comment]
The Readme of @ember/optional-features explicitly mentions that it is only meant to be used in apps, not addons :thinking:

**Comment by Turbo87** [comment]
ah, apparently I missed that it would be added to the dummy app, not the addon itself... 🤦‍♂️

**Comment by rwjblue** [comment]
Yeah, it’s only added as a dev dep in the ember-try scenario. The goal there is to help ensure addons can work in both scenarios as there are some minor differences with event dispatching that _mostly_ don’t matter but sometimes might (and it seems better to have it under test as suggested by the original RFC).

**Comment by cibernox** [comment]
@rwjblue this looks good. Shall we merge?

File: tests/unit/blueprints/addon-test.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ describe('blueprint - addon', function() {
   "devDependencies": {\n\
     "ember-disable-prototype-extensions": "^1.1.2",\n\
     "ember-source-channel-url": "^1.0.1",\n\
-    "ember-try": "^0.2.23",\n\
+    "ember-try": "^1.0.0-beta.3",\n\
```
Comment (kiwiupover) [inline tests/unit/blueprints/addon-test.js]: @rwjblue How far way is ember-try 1.0?

Comment (rwjblue) [reply tests/unit/blueprints/addon-test.js]: We should plan to have 1.0.0 released before this version (ember-cli@3.3.0) is released...

File: blueprints/addon/files/addon-config/ember-try.js [hunk 2]
```diff
@@ -64,6 +64,19 @@ module.exports = function() {
           npm: {
             devDependencies: {}
           }
+        },
+        {
+          name: 'ember-default-with-jquery',
+          env: {
+            EMBER_OPTIONAL_FEATURES: JSON.stringify({
+              'jquery-integration': true,
```
Comment (kellyselden) [inline blueprints/addon/files/addon-config/ember-try.js]: unneeded trailing comma

File: blueprints/addon/files/addon-config/ember-try.js [hunk 3]
```diff
@@ -64,6 +64,19 @@ module.exports = function() {
...
+            }),
```
Comment (kellyselden) [inline blueprints/addon/files/addon-config/ember-try.js]: unneeded trailing comma

**Comment by rwjblue** [comment]
OK, fixed up the linting issues...

**Pull request merged**